src/cqueues.c

```diff
@@ -1982,10 +1982,22 @@ static cqs_status_t cqueue_resume(lua_State *L, struct cqueue *Q, struct callinf
 		}
 	} else {
 		nargs = lua_gettop(T->L);
-		if (status != LUA_YIELD) {
+		if (status == LUA_OK && lua_getstack(T->L, 0, &(lua_Debug){}) > 0) {
```

Comment:
> Why? does >0 vs <= 0 matter?
> ==> What I have is the same as what the lua coroutine library has.

The original code was assuming that if `status !=  LUA_YIELD` then it must be a newly created thread. That's why it was decrementing nargs--to exclude the function at the bottom of the stack (on the first invocation, lua_resume is like lua_pcall). lua_getstack returns 1 if the thread is running. We want to know whether the thread is in the initial state. If you look at the code in lcorolib.c, the non-yielding, suspended condition is only reached if `status == LUA_OK && lua_getstack() <= 0`. But using a condition of  `!= 1` better matches the official documentation for lua_getstack.